pkg/controller/servicebindingrequest/servicebinder.go

```diff
@@ -152,18 +152,6 @@ func (b *serviceBinder) unbind() (reconcile.Result, error) {
 		return requeueError(err)
 	}
 
-	logger.Info("Deleting intermediary secret")
-	if err := b.secret.delete(); err != nil {
-		logger.Error(err, "On deleting intermediary secret.")
-		return requeueError(err)
-	}
-
-	logger.Debug("Removing resource finalizers...")
```

Comment:
I think we still need the finalisers for successful deletion/removal on apps side https://github.com/redhat-developer/service-binding-operator/blob/6f8116b2ac91587cee82f56cdd6c0d911892c4cb/pkg/controller/servicebindingrequest/binder.go#L533, after calling the unbind() https://github.com/redhat-developer/service-binding-operator/blob/master/pkg/controller/servicebindingrequest/servicebinder.go#L150 
Only the secret deletion is being managed by ownerRefs. ownerRefs are not added for apps.

Reply:
Removed this commit

Reply:
@dharmit please clarify if some additional behaviour should be added?

Reply:
No additional behaviour according to me,
This is enough.
Do you have anything specific @pedjak ?

Reply:
> Do you have anything specific @pedjak ?

@Avni-Sharma raised the question above if we still need a finalizer in case when app is removed? Do we need to remove secret if app gets removed?

Reply:
@pedjak @Avni-Sharma 

[1] if we still need a finalizer in case when app is removed?
Yes

[2] Do we need to remove secret if app gets removed?
No